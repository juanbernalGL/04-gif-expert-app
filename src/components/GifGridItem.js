import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = ( {id, title, url} ) => {
    // console.log( props );
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} ></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes= {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};

export default GifGridItem;
