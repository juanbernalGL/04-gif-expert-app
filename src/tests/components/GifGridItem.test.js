import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGridItem from '../../components/GifGridItem';



describe('Verifies that component <GifGridItem /> display correctly', () => {
    const title = 'Compment Grid Item';
    const url = 'https://media1.giphy.com/media/6IkjQmpaRwIabJ2G3C/giphy-downsized-medium.gif?cid=a12b0e9c0s2xapm4oybeuva6a0mh4isidbuhuapdh80g2jtv&rid=giphy-downsized-medium.gif';
    let wrapper = shallow( <GifGridItem  title={title} url={url}/>);
    beforeEach(() => {
         wrapper = shallow( <GifGridItem  title={title} url={url}/>);
    })
    test('Test component <GifGridItem />  create on snapshot', () => {
        // let wrapper = shallow( <GifGridItem title={title} url={url} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('must be contains a p and an image', () => {
        const p = wrapper.find('p').text().trim();
        expect( p ).toBe( title );
    });

    test('must be contains an image, image url and image alt', () => {
        const img = wrapper.find('img');
        // console.log(img.props());
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });
    
    test('must be contain animate__fadeIn class', () => {
        const container = wrapper.find('div');
        // console.log('div', container.props());
        expect( container.prop('className') ).toContain( 'animate__fadeIn' );
        expect( container.prop('className').includes('animate__fadeIn') ).toBe( true );
        // expect( '' ).toBe('animate__fadeIn');
        // expect( '' ).toBe('')
    });
});