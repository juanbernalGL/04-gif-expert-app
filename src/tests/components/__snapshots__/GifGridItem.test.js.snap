// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verifies that component <GifGridItem /> display correctly Test component <GifGridItem />  create on snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Compment Grid Item"
    src="https://media1.giphy.com/media/6IkjQmpaRwIabJ2G3C/giphy-downsized-medium.gif?cid=a12b0e9c0s2xapm4oybeuva6a0mh4isidbuhuapdh80g2jtv&rid=giphy-downsized-medium.gif"
  />
  <p>
    Compment Grid Item
  </p>
</div>
`;
